src/server/dcp_server/models.py

```diff
@@ -60,12 +70,444 @@ def masks_to_outlines(self, mask):
+        # correspond to the classes, to prepare imgs and masks for training CNN
+
+        # TODO I commented below lines. I think we should expect masks to have same size as output of eval, i.e. one channel instances, second channel classes
+        # in this case the shape of masks shall be [2, H, W] or [2, 3, H, W] for 3D
+        # In this case remove commented lines
+        # num_classes = self.model_config["classifier"]["num_classes"]
+        # masks_1channel = [mask.sum(0) for mask in masks]
+        # masks_classifier = [mask if mask.shape[-1] == num_classes else
+                                 # mask.transpose(1, 2, 0) for mask in masks]
+        
+        # train cellpose
+        masks = np.array(masks)
+        masks_instances = list(masks[:,0, ...])
+        self.segmentor.train(imgs, masks_instances)
+        # create patch dataset to train classifier
+        masks_classes = list(masks[:,1, ...])
+        patches, labels = self.create_patch_dataset(imgs, masks_classes, masks_instances)
+        # train classifier
+        self.classifier.train(patches, labels)
+        #return # TODO - define if we need to return something
+
+    def eval(self, img, **eval_config):
+
+        # TBD we assume image is either 2D [H, W] or 3D [H, W, C] (see fsimage storage)
+    
+        # The final mask which is returned should have 
+        # first channel the output of cellpose and the rest are the class channels
+        # TODO test case produces img with size HxW for eval and HxWx3 for train
+        with torch.no_grad():
+            # get instance mask from segmentor
+            instance_mask = self.segmentor.eval(img)
+            # find coordinates of detected objects
+            locs = find_objects(instance_mask) 
+            class_mask = np.zeros(instance_mask.shape)
+            # get patches centered around detected objects
+            patches, _ = self.get_centered_patches(img, 
+                                                   instance_mask, 
+                                                   self.eval_config["classifier"]["data"]["patch_size"], 
+                                                   noise_intensity=5)
+            # loop over patches and create classification mask
+            for idx, patch in enumerate(patches):
+                patch_class = self.classifier.eval(patch)
+                loc = locs[idx]
+                # Assign predicted class to corresponding location in final_mask
+                class_mask[loc] = patch_class.item() + 1
+            # Apply mask to final_mask, retaining only regions where cellpose_mask is greater than 0
+            class_mask = class_mask * (instance_mask > 0)#.long())
+            final_mask = np.stack((instance_mask, class_mask)).astype(np.uint16)
+
+        self.eval_config['z_axis'] = 0
+
+        return final_mask
+
+    # REMOVE? replaced by code in eval
+    '''
+    def get_prediction(self, input_image, cellpose_mask):
```

Comment:
@KorenMary can you check if we can safely remove this function now and remove if so? Thanks!

Reply:
Done